exchanges/orderbook/depth.go

```diff
@@ -323,10 +279,361 @@ func (d *Depth) IsFundingRate() bool {
 	return d.isFundingRate
 }
 
+// GetAskLength returns length of asks
+func (d *Depth) GetAskLength() (int, error) {
+	d.m.Lock()
+	defer d.m.Unlock()
+	if d.validationError != nil {
+		return 0, d.validationError
+	}
+	return d.asks.length, nil
+}
+
+// GetBidLength returns length of bids
+func (d *Depth) GetBidLength() (int, error) {
+	d.m.Lock()
+	defer d.m.Unlock()
+	if d.validationError != nil {
+		return 0, d.validationError
+	}
+	return d.bids.length, nil
+}
+
+// TotalBidAmounts returns the total amount of bids and the total orderbook
+// bids value
+func (d *Depth) TotalBidAmounts() (liquidity, value float64, err error) {
+	d.m.Lock()
+	defer d.m.Unlock()
+	if d.validationError != nil {
+		return 0, 0, d.validationError
+	}
+	liquidity, value = d.bids.amount()
+	return liquidity, value, nil
+}
+
+// TotalAskAmounts returns the total amount of asks and the total orderbook
+// asks value
+func (d *Depth) TotalAskAmounts() (liquidity, value float64, err error) {
+	d.m.Lock()
+	defer d.m.Unlock()
+	if d.validationError != nil {
```

Comment:
Is it okay for the new functions below to be lacking this validation error check?

Reply:
No not at all! 😭